File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let width = 0;
    let height = 0;
    for(let i = 0; i < sizes.length; i++){
        let [a, b] = sizes[i].sort((a, b) => a - b)
        console.log([a, b])
        if(a > width){
            width = a;
        }
        
        if(b > height){
            height = b;
        }
    }

    
    return width * height;
}







/*
function solution(sizes) {
    var w = 0;
    var h = 0;
    for(let i = 0; i < sizes.length; i++){
        let [a, b] = sizes[i].sort((a,b) => a-b);
        if (a > h){
            h = a
        }
        if (b > w){
            w = b
        }
    };

    return w * h;
}

*/

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):

    w = []
    h = []
    
    for s in sizes:
        w.append(max(s))
        h.append(min(s))
        
    return max(w) * max(h)

  # 지갑의 최대 길이, 최소 길이를 각자 나누고
  # 나눈 그룹에서 최대값들을 곱해주면 가장 큰 지갑의 크기를 구할 수 있다.